cloud-energy-collection/iriscasttools/test/test_utils.py

```diff
@@ -0,0 +1,265 @@
+def test_retry(num_fail, expected_calls):
+    """
+    Test "retry" function
+
+    Keyword arguments:
+        num_fail -- int: number of times command to retry is mocked to fail
+        expected_calls -- int: number of times retry function is expected to call subprocess library
+    """
+    mock = Mock()
+    side_effects = [AssertionError for _ in range(num_fail)]
+    side_effects.append("Success")
+
+    mock.side_effect = side_effects
+
+    retry(mock, retry_on=AssertionError, retries=3, delay=3, backoff=2)
+    assert mock.call_count == expected_calls
+
+
+@pytest.mark.parametrize(
+    "valid_path",
+    [
+        ("/dev/ipmi0"),
+        ("/dev/ipmi/0"),
+        ("/dev/ipmidev/0"),
+    ],
+)
+def test_check_ipmi_conn(valid_path):
+    """
+    Test "check_ipmi_conn" function
+
+    Keyword arguments:
+        valid_path: valid path to test
+    """
+
+    def mock_exists(file_path):
+        return str(file_path) == valid_path
+
+    with patch.object(pathlib.Path, "exists", mock_exists):
+        assert check_ipmi_conn()
+
+
+@patch("utils.run_cmd")
+def test_ipmi_raw_power_query(mock_run_cmd):
+    """
+    Test "ipmi_raw_power_query" function
+
+    Keyword arguments:
+        mock_run_cmd -- Mock object for utils.run_cmd function
+    """
+    val = ipmi_raw_power_query()
+    mock_run_cmd.assert_called_once_with(
+        "/usr/sbin/ipmi-dcmi --get-system-power-statistics"
+    )
+    assert val == mock_run_cmd.return_value
+
+
+@pytest.mark.parametrize(
+    "test_args",
+    [
+        (
+            [
+                "current_power",
+                "minimum_power_over_sampling_duration",
+                "maximum_power_over_sampling_duration",
+                "average_power_over_sampling_duration",
+                "time_stamp",
+                "statistics_reporting_time_period",
+                "power_management",
+            ]
+        ),
+        (["some_invalid_arg"]),
+    ],
+)
+@patch("utils.ipmi_raw_power_query")
+@patch("utils.check_ipmi_conn")
+def test_get_ipmi_power_stats(mock_check_ipmi_conn, mock_raw_power_query, test_args):
+    """
+    Test "ipmi_raw_power_query" function
+
+    Keyword arguments:
+        mock_raw_power_query -- mock
+
+    """
+    ipmi_example_fp = os.path.join(_TESTDATA_DIR, "raw_ipmi_test.txt")
+    ipmi_example_exp_vals_fp = os.path.join(_TESTDATA_DIR, "raw_ipmi_test_exp_vals.csv")
+
+    expected_results = {k: "" for k in test_args}
+    raw_query_out, all_expected_vals_dict = read_test_from_file(
+        ipmi_example_fp, ipmi_example_exp_vals_fp
+    )
+
+    mock_raw_power_query.return_value = raw_query_out
+    mock_check_ipmi_conn.return_value = True
+    expected_results.update(
+        {
+            k: v
+            for k, v in all_expected_vals_dict.items()
+            if k in expected_results.keys()
+        }
+    )
+
+    result = get_ipmi_power_stats(*test_args)
+    assert mock_check_ipmi_conn.call_count == 1
```

Comment:
Nit: This can simply be `mock_check_ipmi_conn.assert_called_once()`
As it's more Pythonic